compiler/luci/pass/src/FuseMulWithFullyConnectedPass.cpp

```diff
@@ -118,7 +118,39 @@ bool fuse_mul_with_fc(luci::CircleMul *mul)
   luci::CircleFullyConnected *fc = nullptr;
   luci::CircleConst *multiplication = nullptr;
   RETURN_FALSE_UNLESS(luci::fill(&fc, &multiplication).with_commutative_args_of(mul));
-  // Make sure that FullyConnected has only one successor:
+  /**
+   *  Make sure that FullyConnected has only one successor.
+   *
+   *  If the FullyConnected output is connected to more nodes,
+   *  this pass will replace node with new fused FullyConnected.
+   *  Thus pass success will only introduce extra FullyConnected
+   *  without reducing overall number of nodes.
+   *  Which tends to increase model's size and degrades model's performance.
+   *  Thus one successor is required to benefit from this pass.
+   *
+   *  Example graph that illustrates the described scenario:
```

Comment:
do you have real example for this case?

Reply:
Do you ask for specific models? If yes, I am not able to answer this -- I work of GH issue and searching all of the model zoo for such pattern will take some time.

If you ask about numbers, I ran simple benchmark on these two models (the difference is not "huge" but probably can easily stack when pattern happens multiple times. Please take note that FC is performing not only multiplication but also addition):
![image](https://github.com/user-attachments/assets/5dc8ae75-e78c-478b-8d10-f0ab887b9aec)
^ This gives: ~0.56s on average

![image](https://github.com/user-attachments/assets/a9231d96-9bd7-42e3-9cd5-d567c800c283)
^ This one: ~0.58s on average
(I can share machine settings/setup if necessary)

I can rephrase comment's statement to something like:

> "If the FullyConnected output is connected to more nodes, the pass is returning false.
Otherwise the pass would be designed to copy FullyConnected  node in place of Mul node. Creating extra FullyConnected node that is not reducing overall number of nodes. Such scenario tends to increase model's size and degrades model's performance. To **avoid such replacement**, only one successor is required by this pass. ".

Would you like me to change it?